MT-CTF/capturetheflag
# Improve /report; add report API

- `/report` doesn't really care whether there's a player name in the report or not, but there has to be more than one word.
  - e.g. `/report` and `/report foo` won't work, while `/report foo bar ...` would work.
- The reporter's team is mentioned in IRC reports, and the reporter's name is colourized in in-game reports.
- If the report body contains a player name, that is coloured acc. to the player's team in in-game reports, and their team name added next to their name in IRC reports.
- The above feature requires the `ctf` mod (from [ctf_pvp_engine](https://github.com/rubenwardy/ctf_pvp_engine) modpack) as a dependency, and hence the mod has been renamed `ctf_reports`.
- I've also added an API to `ctf_report`, to allow other mods to report stuff using this API. The `/report` command simply prefixes `<reporter> reported:` to the report body, and passes it to `ctf_report.send_report`.

### Result

- This is what a moderator would see in-game:

  ![screenshot_20190920_132639](https://user-images.githubusercontent.com/36130650/65309941-c5257380-dbaa-11e9-85c2-4fb6614329a5.png)

- This is what a moderator (who's subbed to IRC reports) would see:

  ```
  <ClobberIRC> ClobberXD (team blue) reported: ANAND (team red) Griefing like there's no tomorrow >.> (mods online: ANAND)
  ```

### Testing instructions

It's sufficient to just test the reports themselves, IMO.

- Ensure that there are two players in-game, and one player logged in from IRC. You could just report yourself, I guess, but I recommend reporting another player, preferably on another team to verify whether colourization works as expected.
- Make sure that the IRC player, and at least one player in-game have `kick` and `ban` privs.
- Report the other in-game player; make sure to include their name.
- Observe the report, both in game and on IRC.
